### PR TITLE
Allow user to indicate transport capacity

### PR DESCRIPTION
**Pull request by brucebolt**
Trello: https://trello.com/c/A1S8pkM8/103-indicate-transport-capacity

Adds an "other" option and also a textarea to gather details of the transport that can be offered.  Check your answers page updated to display both fields alongside the same question.

Also refactors the mandatory text field validator to keep it consistent with the other validators.

Screenshot 2020-03-30 at 15 42 35

Screenshot 2020-03-30 at 15 44 08

Trello card: https://trello.com/c/A1S8pkM8

### CODE REVIEW

File: app/controllers/coronavirus_form/business_details_controller.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def submit
   PAGE = "business_details"
 
   def validate_fields(business_details)
-    missing_fields = validate_mandatory_text_fields(REQUIRED_FIELDS, PAGE)
+    missing_fields = validate_mandatory_text_fields(PAGE, REQUIRED_FIELDS)
```
Comment (huwd) [inline app/controllers/coronavirus_form/business_details_controller.rb:34]: alphabetising?

File: app/controllers/coronavirus_form/check_answers_controller.rb [hunk 2]
```diff
@@ -112,6 +116,20 @@ def product_details(products)
     end
   end
 
+  def transport_type(question)
+    answer = []
+    answer << session["transport_type"].flatten.to_sentence
```
Comment (huwd) [inline app/controllers/coronavirus_form/check_answers_controller.rb:121]: [to_sentence is cool](https://apidock.com/rails/Array/to_sentence). Hadn't seen that before!

File: app/views/coronavirus_form/transport_type.html.erb [hunk 3]
```diff
@@ -27,5 +28,15 @@
     }
   end
 } %>
+
+<%= render "govuk_publishing_components/components/textarea", {
```
Comment (huwd) [inline app/views/coronavirus_form/transport_type.html.erb:32]: @bilbof has an open PR to cache some of these, if that gets merged first we should adjust these here too?

**Review by huwd — APPROVED** [review]

File: app/controllers/coronavirus_form/business_details_controller.rb (hunk 1 above)
Comment (brucebolt) [reply app/controllers/coronavirus_form/business_details_controller.rb:34]: Yes.  Also the other validator methods all take page first and the other arguments next.

**Pull request merged**